src/riak_cs_access_archiver.erl

```
@@ -207,9 +207,10 @@ store(User, RcPid, Record, Slice) ->
             ok = lager:debug("Archived access stats for ~s ~p",
                              [User, Slice]);
         {error, Reason} ->
+            IsConnected = riakc_pb_socket:is_connected(MasterPbc),
```

Only a grance at riakc_pb_socket so maybe I'm wrong.
If `auto_reconnect=true`, riakc_pb_socket seems to reply `{error, timeout}` to the caller and dies.
If it is correct, the call `is_connected` results in `noproc` exception and `Reason` is swallowed (not output to log).

This may not cause an issue if the switch `auto_reconnect`, which becomes configurable by this PR, is only for developers and not for production.